text/0256-react-cdk-add-jsx-tsx-support.md

```diff
@@ -0,0 +1,182 @@
+
+Developer accessibility.  Expanding and making the AWS platform accessible
+to a larger group of people.
+
+* Library / framework support is a natural evolution on top of already
+  supported CDK languages.
+* React is familiar to a large group of frontend developers. It's a generalized
+  tech that allows developers to naturally express composition, which is the core
+  tenet of serverless / serviceful solutions.
+* This eases developers into more advanced AWS usage and therefore enabling them
+  to get more value from the platform.
+* Library/framework support on top of already supported language (ts) via jsii
+
+**What use cases does it support?**
+
+The large pool of existing React developers who have experience with React and it's
+component based composition idioms.  Allows them to reuse and leverage their existing
+knowledge to build full stack cloud solutions.
+
+**What is the expected outcome?**
+
+More people leveraging CDK suite -> more innovation and solutions -> advance humanity.
+
+# Design Summary
+
+Code generate a set of ReactCDK components that has feature parity with all TS
+components from cfn constructs to solutions constructs.
+
+# Detailed Design
+
+## Codegen ReactCDK components
+
+A set of ReactCDK components that has feature parity with all TS components from cfn constructs to solutions constructs.
+
+* mapping from react concepts of props, state, children, render props to typescript
+CDK constructors, params, hierarchical nesting (parent), etc.
+* code generated based on the jsii manifests of CDK modules
+[via]((https://twitter.com/emeshbi/status/1305017904027643906?s=20)) @eladb
+* custom react renderer / reconciler where cloud (cfn) is it's render target getting there via CDK language layer.
+* ReactCDKBaseComponent (class or functional component/hooks friendly)
+* ReactCDKPatternComponent - pre-composed infra for common use cases.
+* leverage React ecosystem adjacencies (e.g. react storybook, etc.)
+* wrap these ReactCDK components for Amplify “Backend” library
+* special cases of lambda function code being webpack + babel + etc being
+transpiled, bundled, etc. and handed over to CDK provisioning layer.
+  > this is a bit more involved and also needs research.  this is the blending of app code and infra resources it
+  interacts with.
+  > This is somewhat similar to
+  [Meteor](https://blog.meteor.com/the-meteor-chef-an-introduction-to-apollo-b1904955289e),
+  but instead of only mongodb, it's every cfn supported AWS service.  Need to not make same mistakes as meteor.
+* step function special cases where the components that make it up are customized for developer ergonomics/dx.
+* this is additional alternative to amplify add [backend] (e.g. api, auth, etc.)
+
+# Drawbacks
+
+* React is not universally liked.  If it becomes associated with CDK there is potential brand/product damage.
+* CDK is associated and tied to AWS brand.  AWS tends to support all "popular" frameworks equally.  Favoring one might
```

Comment:
I think that's not a concern. Having support for multiple languages and idioms is actually a tenet of the CDK, and as such this proposal enhances the ecosystem.